### DISCUSSION TITLE
add option to toggle app visibility?

### DISCUSSION THREAD

**Discussion by nikoloz110** [rxhanson/Charmstone-Community#16] [Ideas]
It would be nice to have the possibility to toggle app visibility.  
use case: I have opened 3 main working windows at the same time stacked next to one other: debugger, editor, and simulator.  on the same desktop, I also have the chrome window maximized, for Documentation and google results. I want to have some possibility to toggle between 3 apps and 1 maximized app. In this case, toggling the visibility of Chrome solves my problem.
P.S I also own rectangle pro, and stashing Chrome would solve the problem, but stashing breaks when rectangle window resizing, so can't use that.
P.P.S It's sad that Charmstone does not get the love it deserves. It's a very convenient way to launch apps more effortlessly with less memorization. there is much room for improvement but I understand that for that to happen sales numbers must go up

**Comment by rxhanson** [comment]
Thanks for the feedback.

> It would be nice to have the possibility to toggle app visibility. use case: I have opened 3 main working windows at the same time stacked next to one other: debugger, editor, and simulator. on the same desktop, I also have the chrome window maximized, for Documentation and google results. I want to have some possibility to toggle between 3 apps and 1 maximized app. In this case, toggling the visibility of Chrome solves my problem.

I tried this out before and I prefer the way that it currently works, which is more consistent with macOS. I'll consider adding it as an option, but the app is purposefully slim and is built the way that I desire it to behave. 

> P.S I also own rectangle pro, and stashing Chrome would solve the problem, but stashing breaks when rectangle window resizing, so can't use that.

Can you provide more detail? I would prefer to have stashing work properly and I don't understand exactly what's broken.

> P.P.S It's sad that Charmstone does not get the love it deserves. It's a very convenient way to launch apps more effortlessly with less memorization. there is much room for improvement but I understand that for that to happen sales numbers must go up

Again, the app is built exactly the way that I desire it to work. I add a lot of features into Rectangle Pro because the philosophy of the app is to have one place that provides a lot of different ways I like to manage windows. I enjoy the smaller feature set of Charmstone and see simplicity as a feature, just as it is a feature for Rectangle (non-pro) and many other apps. I'm not closed minded to ever adding features, I'm just really stringent on features that I deem worthy of adding in. That said, I'm interested to hear what you mean by "much room for improvement". Sales numbers aren't really a factor here. If I had much higher sales numbers of Charmstone, I would just assume that more people are trying it out and like it for the same reasons I do.

**Reply by nikoloz110** [reply]
@rxhanson  thanks for the feedback and sorry for the late reply.

here is the issue I opened, regarding the stashing issue, you were interested in.
https://github.com/rxhanson/RectanglePro-Community/issues/135

as for "room for improvement" the app seems very barebones, compared to similar apps, like contexts 3, which has tons of options (though I ended up using only core features of it), giving it a second thought, I can't think of much to add to the charmstone, guess you are correct about keeping it that way. 

though, I will still add a couple of options you might find interesting.
1. possibility to choose from windows of an application  (e.g. when hovering over the chrome icon, a list of chrome windows appears to choose from)
2. option to stick the "charmstone app opener" in the center of a screen. right now it just opens wherever the arrow is currently.
3. option to hide/minimize the app if it is being active, or frontmost ( kind of toggle)
